tests/driver_isl29125/main.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2015 Ludwig Ortmann
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the ISL29125 RGB light sensor
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de>
+ *
+ * @}
+ */
+
+#ifndef TEST_ISL29125_I2C
+#error "TEST_ISL29125_I2C not defined"
+#endif
+
+#include <stdio.h>
+#include <string.h>
+
+#include "vtimer.h"
+#include "isl29125.h"
+
+#define SLEEP       (250 * 1000U)
+
+int main(void)
+{
+    isl29125_t dev;
+    isl29125_rgb_t data;
+    memset(&data, 0x00, sizeof(data));
+
+    puts("ISL29125 light sensor test application\n");
+    printf("Initializing ISL29125 sensor at I2C_%i... ", TEST_ISL29125_I2C);
+    if (isl29125_init(&dev, TEST_ISL29125_I2C, ISL29125_I2C_ADDRESS,
+                ISL29125_MODE_RGB, ISL29125_RANGE_10K,
+                ISL29125_RESOLUTION_16) == 0) {
+        puts("[OK]\n");
+    }
+    else {
+        puts("[Failed]");
+        return 1;
+    }
+
+    /* try out some modes */
+    puts("Setting mode to RGB"); /* done above */
+    if (isl29125_read_rgb(&dev, &data) == -1) {
+        puts("Error reading data, terminating.");
+        return 1;
+    }
+    printf("RGB value: (%5i / %5i / %5i) lux\n", (int)data.red, (int)data.green, (int)data.blue);
+
```

Comment:
I like the idea of testing different modes but I agree with @haukepetersen that a loop could be easier to read